link/src/main/java/com/stripe/android/link/ui/verification/VerificationViewModel.kt

```diff
@@ -0,0 +1,103 @@
+package com.stripe.android.link.ui.verification
+
+import android.app.Application
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.core.Logger
+import com.stripe.android.core.injection.Injectable
+import com.stripe.android.core.injection.injectWithFallback
+import com.stripe.android.link.LinkActivityContract
+import com.stripe.android.link.LinkScreen
+import com.stripe.android.link.account.LinkAccountManager
+import com.stripe.android.link.injection.DaggerLinkViewModelFactoryComponent
+import com.stripe.android.link.injection.LinkViewModelSubcomponent
+import com.stripe.android.link.model.Navigator
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Provider
+
+/**
+ * ViewModel that handles user verification confirmation logic.
+ */
+internal class VerificationViewModel @Inject constructor(
+    private val linkAccountManager: LinkAccountManager,
+    private val navigator: Navigator,
+    private val logger: Logger
+) : ViewModel() {
+
+    val linkAccount by lazy {
+        // The Link account is already loaded in linkAccountManager when we get to Verification
+        requireNotNull(linkAccountManager.linkAccount)
+    }
+
+    fun onVerificationCodeEntered(code: String) {
+        viewModelScope.launch {
+            linkAccountManager.confirmVerification(code).fold(
+                onSuccess = {
+                    navigator.navigateTo(LinkScreen.Wallet)
+                },
+                onFailure = ::onError
+            )
+        }
+    }
+
+    fun onResendCodeClicked() {
+    }
```

Comment:
Should this have a TODO?

Reply:
Implemented here since it's a small change.